jax_distribution/types.py

```diff
@@ -43,6 +43,18 @@ class LearnerState(NamedTuple):
     timestep: TimeStep
 
 
+class RNNLearnerState(NamedTuple):
+    """State of the `Learner` for recurrent architectures."""
+
+    params: FrozenDict
+    opt_state: OptState
+    key: chex.PRNGKey
+    env_state: LogEnvState
+    timestep: TimeStep
+    dones: chex.Array
+    hstate: chex.Array
```

Comment:
Something that could be worth look at is having this be something like 
`hstates: Union[chex.Array, Tuple[chex.Array, chex.Array]]`
This could enable use to reuse the same `NamedTuple` for rec ippo and rec mappo. It might be a bit confusing in the rec ippo case to the have a plural word here when we only have a single hidden state though. Keen to hear your and @sash-a's thoughts on this.